### DISCUSSION TITLE
deleted

### DISCUSSION THREAD

**Discussion by ghost** [LSPosed/LSPosed#476] [Q&A] [answered]
deleted

**Comment by yujincheng08** [comment]
Your log is too late. Native modules are loaded right after loading the module, and there will be a log saying
```
native_api: Registered xxx.so
```

Next time, for questing, plz go to the discussion. I will move this issue to the discussion.

**Comment by ghost** [comment]


**Reply by yujincheng08** [reply]
`native_api: Registered liblsposed_module.so`

your library has been registered. Have you loaded it?

**Reply by yujincheng08** [reply]
We won't load it for you. You should call `System.loadLibrary` on `handleLoadPackage`.

**Reply by ghost** [reply]
Ok, so it's load in the app and so I can hook. Good I will try.

**Comment by yujincheng08** [comment, marked as answer]
1. `native_init` only tells LSPosed what library should call `native_init`, but we won't load it for you. So you have to load it at the proper time (but not in `initZygote`).
2. You should hook `_ZN7_JNIEnv9FindClassEPKc` on `JNI_ONLoad` by `hook_func(&env->FindClass, &my_findClass, &bk_findClass);` because it's not any function on the app lib.

**Reply by yujincheng08** [reply]
Why didn't you read my sentence:
> because it's not any function on the app lib

**Reply by yujincheng08** [reply]
? You lack some basic pieces of knowledge. It's hard to communicate with you... I have told you the exact solution. So I won't reply to this question again.
